### ISSUE TITLE
从AlexNet模型（正常跑通） 切换到ResNet50 模型时报错 

### ISSUE THREAD

**Issue by liyutg** [PaddlePaddle/models#2046]
**因为使用了自定义数据集，所以修改了读取数据的代码**
### 系统信息
paddle 版本 1.3.2
GPU 型号
Tesla K40m 11441MiB 
Please NOTE: device: 1, CUDA Capability: 35, Driver API Version: 10.1, Runtime API Version: 9.0
device: 0, cuDNN Version: 7.0.

### 项目配置信息
```
-------------  Configuration Arguments -------------
      --model=ResNet50 \
      --batch_size=256 \
      --total_images=31718 \
      --class_dim=61 \
      --image_shape=3,224,224 \
      --model_save_dir=output/ \
      --with_mem_opt=True \
      --lr_strategy=piecewise_decay \
	  --num_epochs=120 \
      --lr=0.1 \
      --l2_decay=1e-4 \
----------------------------------------------------
```
### 报错信息
W0413 13:19:44.940369 13978 device_context.cc:263] Please NOTE: device: 0, CUDA Capability: 35, Driver API Version: 10.1, Runtime API Version: 9.0
W0413 13:19:44.940452 13978 device_context.cc:271] device: 0, cuDNN Version: 7.0.
W0413 13:19:50.571400 16109 system_allocator.cc:122] Cannot malloc 9999.6 MB GPU memory. **Please shrink FLAGS_fraction_of_gpu_memory_to_use environment variable to a lower value. Current value is 0.92**
W0413 13:19:50.571779 16109 legacy_allocator.cc:191] Cannot allocate 196.000000MB in GPU 0, available 1.214539GB
W0413 13:19:50.571794 16109 legacy_allocator.cc:194] total 11996954624
W0413 13:19:50.571805 16109 legacy_allocator.cc:195] GpuMinChunkSize 256.000000B
W0413 13:19:50.571813 16109 legacy_allocator.cc:198] GpuMaxChunkSize 9.765232GB
W0413 13:19:50.571827 16109 legacy_allocator.cc:201] GPU memory used: 9.633711GB
*** Aborted at 1555132790 (unix time) try "date -d @1555132790" if you are using GNU date ***

**Comment by liyutg** [comment]
## FLAGS_fraction_of_gpu_memory_to_use
### 改成 0.2之后
```
W0413 23:19:05.162389 17655 device_context.cc:271] device: 0, cuDNN Version: 7.0.
W0413 23:19:07.378151 17745 system_allocator.cc:122] Cannot malloc 2173.83 MB GPU memory. Please shrink FLAGS_fraction_of_gpu_memory_to_use environment variable to a lower value. Current value is 0.2
W0413 23:19:07.378525 17745 legacy_allocator.cc:191] Cannot allocate 98.000000MB in GPU 0, available 373.937500MB
W0413 23:19:07.378542 17745 legacy_allocator.cc:194] total 11996954624
W0413 23:19:07.378552 17745 legacy_allocator.cc:195] GpuMinChunkSize 256.000000B
W0413 23:19:07.378561 17745 legacy_allocator.cc:198] GpuMaxChunkSize 2.122877GB
W0413 23:19:07.378576 17745 legacy_allocator.cc:201] GPU memory used: 10.508847GB
*** Aborted at 1555168747 (unix time) try "date -d @1555168747" if you are using GNU date ***
```
### 改成 0.99后
```
C++ Callstacks: 
Enforce failed. Expected allocating <= available, but received allocating:112831358240 > available:11397106893.
Insufficient GPU memory to allocation. at [/paddle/paddle/fluid/platform/gpu_info.cc:216]
PaddlePaddle Call Stacks: 
0       0x7f12da02837dp void paddle::platform::EnforceNotMet::Init<std::string>(std::string, char const*, int) + 365
1       0x7f12da0286c7p paddle::platform::EnforceNotMet::EnforceNotMet(std::string const&, char const*, int) + 87
2       0x7f12dbbbb4d8p paddle::platform::GpuMaxChunkSize() + 744
3       0x7f12dbbb1c89p
4       0x7f13295d8a80p pthread_once + 80

```

### 改成0之后
```
W0413 23:19:07.378151 17745 system_allocator.cc:122] Cannot malloc 2173.83 MB GPU memory. Please shrink FLAGS_fraction_of_gpu_memory_to_use environment variable to a lower value. Current value is 0.2
W0413 23:19:07.378525 17745 legacy_allocator.cc:191] Cannot allocate 98.000000MB in GPU 0, available 373.937500MB
W0413 23:19:07.378542 17745 legacy_allocator.cc:194] total 11996954624
W0413 23:19:07.378552 17745 legacy_allocator.cc:195] GpuMinChunkSize 256.000000B
W0413 23:19:07.378561 17745 legacy_allocator.cc:198] GpuMaxChunkSize 2.122877GB
W0413 23:19:07.378576 17745 legacy_allocator.cc:201] GPU memory used: 10.508847GB
*** Aborted at 1555168747 (unix time) try "date -d @1555168747" if you are using GNU date ***
```

**Comment by liyutg** [comment]
调小了batch_size 为32 可以运行
但是不知道内存怎么计算出来的
